capstone-frontend/src/components/ListPage.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+
+class ListPage extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      gBooks: [],
+      loading: true,
+      empty: false
+    }
+  }
+
+
+  fetchBooks = async () => {
+
+    const userID = window.localStorage.getItem("userID");
+
+    console.log("Fetch Books LIST PAGE")
+
+    const bookLists = await fetch(`/api/booklist?userID=${userID}`, {
+      method: "GET",
+    }).then(resp => resp.json());
+
+    console.log("Booklist", bookLists)
+
+    const bookList = Array.from(bookLists).filter(bookList => (bookList.id === this.props.match.params.id))[0].gbookIDs;
+
+    console.log("Booklist", bookList)
+
+    if (bookList.length === 0) {
+      this.setState({loading: false, empty: true});
+      return;
+    }
+
+    const gBooks = [];
+
+    await Promise.all(bookList.map(async (gBookID) => {
+      const gBook = await fetch(`https://www.googleapis.com/books/v1/volumes/${gBookID}`).then(response => response.json())
+      gBooks.push(gBook);
+    }))
+
+    this.setState({ gBooks, loading: false });
+    console.log(gBooks)
+  }
+
+  componentDidMount() {
+    this.fetchBooks();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.match.params.id !== prevProps.match.params.id) {
```

Comment:
I used this same logic on my ClubPage, and found that it was repeatedly fetching to no end. Not sure if it is caused by this logic, or something else unique to the ClubPage but you should take a look before you merge.

Reply:
Interesting, I haven't those issues. I will do more testing before merging.

Reply:
I did get issues but they are now fix.